### ISSUE TITLE
[Feature Request]:  Crossed contrasts in multi-factor multi-level ANOVA

### ISSUE THREAD

**Issue by bfmalle** [jasp-stats/jasp-issues#2025]
### Description

Multi-factorial multi-level ANOVAs are difficult to interpret unless one defines meaningful contrasts for each main effect AND crosses these contrasts for the interaction term.  Currently, JASP's interaction contrasts treat the cells of the interaction as if they were one big factor. Instead, we should be able to cross the defined contrasts and be able to interpret the "sub-interactions"

### Purpose

Meaningful contrasts to interpret complex designs 

### Use-case

Any multi-factorial design where at least one factor has more than one df 

### Is your feature request related to a problem?

The problem is that we cannot currently break down interactions (e.g., in a 3 x 3 design) into meaningful patterns.  This isn't a statistical problem but a practical, interpretational problem. 

### Is your feature request related to a JASP module?

ANOVA

### Describe the solution you would like

We currently are able to define contrasts for each factor in a design (e.g., 3 x 2), but these contrasts are only useful for testing the main effects of these factors.  If we want to "apply" the contrasts to the (say, 3 x 2) interaction term, JASP doesn't cross the defined contrasts (A1 x B, A2 x B) but asks us to define a new contrasts that then treats all 2 x 3 cells as if the were one big 1 x 6 factor.

### Describe alternatives that you have considered

Allow user do define a contrast for each factor in the model and then automatically cross these contrasts for the interaction terms.   For example, if A has 3 levels and B has 2 levels, then we might define two Helmert contrasts for A (A1v23, A2v3) and B is of course just a difference contrast B1v2).  The interaction would then consist of two crossed terms:  A1v23 x B12 and A2v3 x B12.  These are meaningful interaction contrasts that can be easily tested, graphed, and interpreted.   

### Additional context

I am attaching a zipped JASP file for a 3 x 3 design, which produces the strange "interaction contrasts."  I also include a printout of an SPSS output in which I defined Helmert contrasts for one (2-level) factor and Polynomial contrasts for the other (2-level factor), and the program automatically crosses the two contrasts so that I have 4 "subcontrasts."
[3 x 3 design example.jasp.zip](https://github.com/jasp-stats/jasp-issues/files/10775300/3.x.3.design.example.jasp.zip)
[3x3ANOVA with crossed contrasts SPSS.pdf](https://github.com/jasp-stats/jasp-issues/files/10775303/3x3ANOVA.with.crossed.contrasts.SPSS.pdf)

**Comment by tomtomme** [comment]
@bfmalle 
I am unsure, if I understand the problem. 
Your file has a 3x3 design that enable 9 pairwise comparisons - this is what JASPs produces.
But since you produced Helmert contrasts for the two main effects, this effectively reduces the available comparisons to a 2x2 matrix. This is what SPSS puts out in the last table seen in the PDF.
Is this correct so far?
So to implement this, our interaction contrast table would need to be conditional on the tables produced for the main effect contrasts.

@JohnnyDoorn 
Any idea if this is feasible? 
Also: Could @bfmalle use custom contrasts as a workaround?

**Comment by bfmalle** [comment]
Yes, this is correct.  There are only (3-1) x (3-1) = 4 dfs for the interaction contrasts, represented by your "2 x 2" matrix.  Pairwise comparisons don't test the interactions.  Interactions are logically tied to the main effects, so ideally they would "continue" the contrasts defined for the main effects.  

I have been using custom contrasts, but honestly, it's a pain, can lead to inaccuracies, and gets worse with a 3-factor design...

**Comment by JohnnyDoorn** [comment]
Hi @bfmalle,

Thanks for the detailed request - this would be great to have. If I understand correctly, it's a sort of "simple contrasts" - like "simple main effects" is an anova conditional on each level of a factor, this would be contrast analysis for each (contrast-)level of a factor? I will look into this because I think it's one of the last "bigger" features from SPSS that is not in JASP's ANOVA yet. 

I'm having some trouble wrapping my head around the SPSS output though, especially the last table - could you help me interpret some of its metrics? For instance the first row: 
parameter = 2 
coefficient = 7.9667364014 
SE = 1.63217 
t =- 4.88106 
p = .00001
What does this particular t/p-value tell me? Or am i I even looking at the wrong table here?

**Comment by bfmalle** [comment]
No, it's the regular set of crossed interaction contrast, as shown in the blue and green columns below.  And the parameter table shows the first two contrasts (main effects for Dose), tested for significance.  These contrasts are also shown in the ANOVA table, as the individual single-df tests for Dose.  These are the same as the ones Jasp shows in the Contrast table for Dose.  What's missing in Jasp are the tests for the four interaction contrasts, shown in the second image below (with green corresponding to the four green contrast weight schemes in the first image).  Confusingly, SPSS numbers the "parameters" differently in the ANOVA table and in the individual contrast (t) tests.   

  
<img width="828" alt="screenshot_2895" src="https://github.com/jasp-stats/jasp-issues/assets/53263125/57114e33-e392-435e-977b-f2a484e675ad">

**Comment by bfmalle** [comment]
Hmm, the second image wasn't displayed in the previous email.  Here it is:

<img width="831" alt="screenshot_2896" src="https://github.com/jasp-stats/jasp-issues/assets/53263125/2da2ac82-7c3e-458f-9ecc-45096e02748e">

**Comment by bfmalle** [comment]
And, as a reminder, the ANOVA table is here.  You see how the four tests correspond precisely to each other (where F = t^2)
<img width="703" alt="screenshot_2897" src="https://github.com/jasp-stats/jasp-issues/assets/53263125/c9c58f37-f1ad-44cf-9744-0907cf838e2d">

**Comment by tomtomme** [comment]
@bfmalle 
Thanks for the further details.

**Comment by bfmalle** [comment]
Hello again, I see that you still have the unusual interaction contrasts in the latest version (19.1).   I'd like to make one more case for a more standard set of interaction contrasts. 

I am attaching a JASP file in which I ran a 3 x 2 ANOVA, asking for polynomial contrasts on the 3 x 2 interaction (Time [1,2,3] x Justification [Yes, No]). The three-level factor happens to be repeated measures, but the same occurs in a fully between-subjects design.  I am also attaching an Excel file in which I illustrate the JASP contrasts and what I would consider the standard polynomials. Let me briefly explain my point verbally, then the Excel file will make more sense.

My understanding of contrasts is that one typically limits comparisons to k - 1 contrasts (often orthogonal, but not necessarily).  So for a 3 x 2 interaction, with 2 dfs, that would be two such polynomial contrasts, a linear and a quadratic one.  The linear interaction contrast would compare the positive linear time trend of the "No" condition (-0.5, 0, +0.5) against the negative linear trend of the "Yes" condition (+0.5, 0, -0.5).  The quadratic interaction contrast  would compare the positive quadratic trend of the "No" condition against the negative quadratic trend of the "Yes" condition.  Given that I chose an orthogonal contrast set, it makes a lot of sense that the two interaction contrasts exhaust the SSQs of the omnibus interaction.  This is what SAS and SPSS have done for a long time.  The beauty of this approach is also that the univariate SSQs nicely sum up for orthogonal contrasts.  So in the example, SSQ for the (2-df omnibus) interaction is 2.66, and the sum of the linear and quadratics interaction contrasts is  =  2.23 + 0.43 = 2.66. (See attached SPSS output.)
[Archive.zip](https://github.com/user-attachments/files/17955103/Archive.zip)


What JASP does instead is **reorganize the 3 x 2 design into a 1 x 6 design** and then create five polynomials over those six cells. But these are not tests of the **3 x 2 interaction** contrast. They are tests for a different kind of design altogether.  (The same happens when I ask for Helmert or any other contrast pattern.)

When I first tried to make sense of what's going on, I reorganized the JASP contrast coefficient tables to see what these comparisons do to the **actual 3 x 2** design —  this is what the Excel file shows.  The five comparisons represent a weird collection of patterns that decompose the data in multiple, correlated ways, and none of them include the standard linear and quadratic interaction contrasts.  I think it's highly confusing, many people won't know that you reorganized the 3 x 2 into a 1 x 6 design, and it doesn't give us the actual two interaction contrasts for the 3 x 2 design.  

I know I can create my own custom contrasts, and I did so in the second run in the JASP file.  Those two contrasts are exactly what SPSS shows in its /CONTRAST([_factor_]) = POLYNOMIAL command.

My request is to make the standard interaction contrasts for the design at hand available and leave cell reorganizations to a custom option.    

Best,

Bertram




JASP Version: JASP 0.19.1 (Apple Silicon)
Build Branch: development
Build Date: Sep  9 2024 13:50:55 (Netherlands)
Last Commit: 28c9e9916b15ff84dfb998ad5db78a421b985aa0

-------- Basic Info --------
Operating System: macOS 14.7
Product Version: 14.7
Kernel Type: darwin
Kernel Version: 23.6.0
Architecture: arm64
Install Path: /Applications/JASP.app/Contents/MacOS
Platfotm Name: cocoa
System Local: en_US

**Comment by bfmalle** [comment]
[Archive.zip](https://github.com/user-attachments/files/17955117/Archive.zip)

Not sure the zip file came through above. Here it is again.
